### ISSUE TITLE
Tray icon is missing

### ISSUE THREAD

**Issue by fchiacchiaretta** [flathub/chat.rocket.RocketChat#12]
Hi,
I'm running Fedora 29 Workstation.
Despite being enabled in the application, tray icon does not show in KStatusNotifier/AppIndicator gnome shell extension. The same application installed from RPM package properly shows the icon.
As far as I understand, flatpak package should bundle libappindicator and libdbusmenu and add needed dbus interface, the same has been done in [this commit](https://github.com/flathub/io.github.Pithos/commit/0cf4ba1eed8dd662fdc4cc1da091b43e7897b73d) for io.github.Pithos.

Best,
Federico Chiacchiaretta

**Comment by andrevmatos** [comment]
Hmm, weird. I use plasma, which also uses libappindicator, and it have been working perfectly here. Do you see any warnings or erros when running with `flatpak run --env=ELECTRON_ENABLE_LOGGING=1 chat.rocket.RocketChat` from the terminal?

**Comment by tassoevan** [comment]
Since 2.14.0 tray icon is disabled by default in Linux environments. Our motivation was the absence of a method via Electron API to detect if the desktop environment support a system tray or not. To enable it, check the menu item `View > Tray icon`. I apologize for the inconvenience of not providing a way to keep the setting on update.

**Comment by fchiacchiaretta** [comment]
Hi @tassoevan,
thanks for the clarification. I understand the issue with the absence of API, system tray is kind of mess in Linux.
Of course I have tray icon option enabled on my client, I reported the bug here in flathub because tray icon properly shows if I install the application via official RPM (tested again today with latest version, exporting XDG_CURRENT_DESKTOP="ubuntu:GNOME" on launch, as suggested [here](https://github.com/electron/electron/issues/10427#issuecomment-386024975)).

**Comment by fchiacchiaretta** [comment]
I tried to do some debugging on dbus with [d-feet](https://github.com/GNOME/d-feet), with both flatpak and non-flatpak version of application.
Watching org.kde.StatusNotifierItem interface (on session bus) I can find that RPM version registers under "MatchRules" (and icon properly shows):

```
type='signal',interface='org.kde.StatusNotifierItem',path='/org/ayatana/NotificationItem/RocketChat1',sender=':1.359'
```

while flatpak version doesn't.

**Comment by tassoevan** [comment]
Kinda similar issue: https://github.com/RocketChat/Rocket.Chat.Electron/issues/1109

**Comment by andrevmatos** [comment]
@fchiacchiaretta just pushed 2.15.1 which go back to pre-built packages since I was having too much trouble with electron4 + node10.11.0 symbols. Can you test again, please?

**Comment by lasombra** [comment]
As soon as it gets updated on Flathub I'll give it a try

**Comment by fchiacchiaretta** [comment]
@andrevmatos just tested updated flatpak (latest version 2.15.1) and tray icon still does not show.
RPM version works as expected (tested latest version with XDG_CURRENT_VERSION workaround).

**Comment by lasombra** [comment]
That's working now, thanks!

P.S.: The tray icon looks very low res, but I don't know if that's up to you or the Rocket.Chat folks

**Comment by fchiacchiaretta** [comment]
@lasombra What DE are you on? What distro?
I'm on Gnome 3.30.2 on Fedora 29 and it still does not work with latest version.

**Comment by lasombra** [comment]
@fchiacchiaretta Fedora 29, KDE Plasma 5.14.5

**Comment by niriak** [comment]
On Linux Mint 19 with latest flatpak version 1.5.1 it doesn't work either - same grey icon shown for every status. And it's looks like the icon is some pixels to big for the tray.

**Comment by fchiacchiaretta** [comment]
I finally managed to get it working.
Looking at [Franz flathub repo](https://github.com/flathub/com.meetfranz.Franz) I did the following changes to chat.rocket.RocketChat.json and rebuild it with flatpak-builder:

1. Switched to base org.electronjs.Electron2.BaseApp version 18.08
2. Added "--talk-name=org.kde.StatusNotifierWatcher" to finish-args
3. Added a bash script which executes
`exec env TMPDIR=$XDG_CACHE_HOME /app/Rocket.Chat/rocketchat-desktop "$@"`
and use that script in desktop file.

I could not get it working with old base io.atom.electron.BaseApp, and point number 3 was necessary to properly show the icon on tray, otherwise I get a menu with an invisible icon.

I hope this can help, check [Franz flathub repo](https://github.com/flathub/com.meetfranz.Franz) for reference :)
